From 6475ea889e7ae22942ce424a36963ac90b02cef1 Mon Sep 17 00:00:00 2001
From: Bryan Cantrill <bryan@joyent.com>
Date: Wed, 28 Feb 2018 00:51:39 +0000
Subject: [PATCH] OS-6705 thread spinning in dp_pcache_poll() induces system
 hang Reviewed by: Dan McDonald <danmcd@joyent.com> Reviewed by: Patrick
 Mooney <patrick.mooney@joyent.com> Approved by: Patrick Mooney
 <patrick.mooney@joyent.com>

---
 usr/src/uts/common/io/devpoll.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/common/io/devpoll.c b/usr/src/uts/common/io/devpoll.c
index 040a1f9190..e0d37e171b 100644
--- a/usr/src/uts/common/io/devpoll.c
+++ b/usr/src/uts/common/io/devpoll.c
@@ -25,7 +25,7 @@
 
 /*
  * Copyright (c) 2012 by Delphix. All rights reserved.
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -405,6 +405,7 @@ repoll:
 					 * about the fd reassignment.
 					 */
 					pdp->pd_fp = fp;
+					pdp->pd_gen = gen;
 				}
 			}
 			/*
-- 
2.21.0

